internal/resources/otelcollector/gateway_test.go

```diff
@@ -2,613 +2,158 @@ package otelcollector
-			DynamicCPURequest:    dynamicCPURequest,
-			BaseCPULimit:         baseCPULimit,
-			DynamicCPULimit:      dynamicCPULimit,
-			BaseMemoryRequest:    baseMemoryRequest,
-			DynamicMemoryRequest: dynamicMemoryRequest,
-			BaseMemoryLimit:      baseMemoryLimit,
-			DynamicMemoryLimit:   dynamicMemoryLimit,
+		{
+			name:           "trace gateway",
+			sut:            NewTraceGatewayApplierDeleter(image, namespace, priorityClassName),
+			goldenFilePath: "testdata/trace-gateway.yaml",
 		},
-	}
-}
-
-func createGatewayRBAC() Rbac {
-	clusterRole := &rbacv1.ClusterRole{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      gatewayName,
-			Namespace: gatewayNamespace,
-			Labels:    labels.MakeDefaultLabel(gatewayName),
+		{
+			name:           "log gateway",
+			sut:            NewLogGatewayApplierDeleter(image, namespace, priorityClassName),
+			goldenFilePath: "testdata/log-gateway.yaml",
 		},
-		Rules: []rbacv1.PolicyRule{
-			{
-				APIGroups: []string{"test"},
-				Resources: []string{"test"},
-				Verbs:     []string{"test"},
-			},
+		{
+			name:           "metric gateway with istio",
+			sut:            NewMetricGatewayApplierDeleter(image, namespace, priorityClassName),
+			istioEnabled:   true,
+			goldenFilePath: "testdata/metric-gateway-istio.yaml",
 		},
 	}
 
-	clusterRoleBinding := &rbacv1.ClusterRoleBinding{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      gatewayName,
-			Namespace: gatewayNamespace,
-			Labels:    labels.MakeDefaultLabel(gatewayName),
-		},
-		Subjects: []rbacv1.Subject{{Name: gatewayName, Namespace: gatewayNamespace, Kind: rbacv1.ServiceAccountKind}},
-		RoleRef: rbacv1.RoleRef{
-			APIGroup: "rbac.authorization.k8s.io",
-			Kind:     "ClusterRole",
-			Name:     gatewayName,
-		},
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var objects []client.Object
+
+			scheme := runtime.NewScheme()
+			utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+			utilruntime.Must(istiosecurityclientv1.AddToScheme(scheme))
+
+			client := fake.NewClientBuilder().WithScheme(scheme).WithInterceptorFuncs(interceptor.Funcs{
+				Create: func(_ context.Context, c client.WithWatch, obj client.Object, _ ...client.CreateOption) error {
+					objects = append(objects, obj)
+					// Nothing has to be created, just add created object to the list
+					return nil
+				},
+			}).Build()
+
+			err := tt.sut.ApplyResources(context.Background(), client, GatewayApplyOptions{
+				AllowedPorts:        []int32{5555, 6666},
+				CollectorConfigYAML: "dummy",
+				CollectorEnvVars: map[string][]byte{
+					"DUMMY_ENV_VAR": []byte("foo"),
+				},
+				IstioEnabled: tt.istioEnabled,
+				Replicas:     2,
+			})
+			require.NoError(t, err)
+
+			if tt.saveGoldenFile {
+				testutils.SaveAsYAML(t, scheme, objects, tt.goldenFilePath)
+			}
+
+			bytes, err := testutils.MarshalYAML(scheme, objects)
+			require.NoError(t, err)
+
+			goldenFileBytes, err := os.ReadFile(tt.goldenFilePath)
+			require.NoError(t, err)
```

Comment:
im not a big fan of the comparison of those two ginormous strings. 
If we do not implement comparison of the objects, I assume it would be already a big help to finding issues when we at least split the big string in per object strings. That way we can not only print the issue, but also which object was affected.